packages/fxa-content-server/app/scripts/views/pair/index.js

```diff
@@ -21,22 +22,27 @@ class PairIndexView extends FormView {
     const uap = this.getUserAgent();
     const isFirefoxDesktop = uap.isFirefoxDesktop();
 
-    if (! isFirefoxDesktop) {
-      // other browsers show an unsupported screen
+    if (! isFirefoxDesktop || ! this.broker.hasCapability('supportsPairing')) {
+      // other browsers show an unsupported screen or if no capability to pair
       return this.replaceCurrentPage('pair/unsupported');
     }
 
+    const account = this.broker.get('browserSignedInAccount');
     // If we reach this point that means we are in Firefox Desktop
-    if (! this.broker.get('browserSignedInAccount')) {
-      // if we are not logged into Sync then we offer to sign in
+    if (! account) {
+      // if we are not logged into Sync then we offer to connect
       return this.replaceCurrentPage('connect_another_device');
     }
 
-    if (! this.broker.hasCapability('supportsPairing')) {
-      return this.replaceCurrentPage('pair/unsupported');
+    if (! account.verified || ! account.sessionToken) {
+      // if account is not verified or missing sessionToken then offer to sign in or confirm
+      return this.navigateAway(this.getEscapedSyncUrl('signin', 'fxa:pair'));
     }
 
-    return this.checkTotpStatus();
+    // here we pass a token from the 'browserSignedInAccount'. This token has a device
+    // attached to it and is the one we want to check. We want to avoid using the
+    // "web" session token because it may disappear, expire or get pruned.
```

Comment:
I see what's going on now answering the question of why. What might be conceptually simpler is to continue to use the getSignedInAccount that's stored in the user, but to avoid the problems you mention here, update [app-start.js->_getUserStorageInstance](https://github.com/mozilla/fxa/blob/74d4d8e15160dc3524788d2c5c313971ce2d867e/packages/fxa-content-server/app/scripts/lib/app-start.js#L575) to use the memory storage if you are pairing, and update [user.js->shouldSetSignedInAccountFromBrowser](https://github.com/mozilla/fxa/blob/74d4d8e15160dc3524788d2c5c313971ce2d867e/packages/fxa-content-server/app/scripts/models/user.js#L725) if you are pairing.